### ISSUE TITLE
Using custom style and config results in one being ignored.

### ISSUE THREAD

**Issue by n0mn0m** [markdownlint/markdownlint#343]
## Description
I am trying to combine use of a config file and style file, however only one seems to be picked up, the settings are not combined.

### Environment
OS X Catalina zsh

**MDL Version**
- 0.10.0

### Expected Behavior
mdl uses `.mdlrc` to enable `ignore_front_matter` and uses the custom styles in `mdl.rb` when checking styles.

```
 mdl --style=mdl.rb -c=.mdlrc input/posts/dataclasses-as-config-objects.md
```

### Actual Behavior
Custom styles are used, but the config is ignored.

## Replication Case

Using the custom config file:

```
mdl -c=.mdlrc input/posts/dataclasses-as-config-objects.d
input/posts/dataclasses-as-config-objects.md:57: MD029 Ordered list item prefix
input/posts/dataclasses-as-config-objects.md:62: MD029 Ordered list item prefix
input/posts/dataclasses-as-config-objects.md:68: MD029 Ordered list item prefix
```

Using the custom style file:

```
mdl --style=mdl.rb input/posts/dataclasses-as-config-objects.md
input/posts/dataclasses-as-config-objects.md:1: MD041 First line in file should be a top level header
input/posts/dataclasses-as-config-objects.md:29: MD046 Code block style
input/posts/dataclasses-as-config-objects.md:40: MD046 Code block style
```

Trying to combine them:

```
mdl -c=.mdlrc --style=mdl.rb input/posts/dataclasses-as-config-objects.md
input/posts/dataclasses-as-config-objects.md:1: MD041 First line in file should be a top level header
input/posts/dataclasses-as-config-objects.md:29: MD046 Code block style
input/posts/dataclasses-as-config-objects.md:40: MD046 Code block style
```

Looks like the config gets ignored and just the style file is used. Trying to use both to enable ignoring front matter then providing my style rules.

.mdlrc
```text
ignore_front_matter true
```

mdl.rb
```ruby
all
rule 'MD013', :tables => false
rule 'MD013', :img => false
rule 'MD013', :code_blocks => false
rule 'MD024', :allow_different_nesting => true
rule 'MD029', :style => 'ordered'
rule 'MD046', :style => 'fenced'
```

Example markdown file:

Example .md blog post:

```markdown
Title: Using Dataclasses for Configuration
Published: 2019-06-23
Tags:

- python
- dataclasses
- 3.7

---

# What are dataclasses

Introduced in Python 3.7 `dataclasses` are normal Python classes with some
extra features for carrying around data and state. If you find yourself writing
a class that is mostly attributes it's a dataclass.

Dataclasses have some other nifty features out of the box such as default
double underscore methods, type hinting, and more.

For more information checkout the
[docs](https://docs.python.org/3/library/dataclasses.html).

## Dataclasses as configuration objects

Recently I've had the opportunity to work on a couple of Python 3.7 projects.
In each of them I was interacting with many databases and API Endpoints.
Towards the beginning of one of the projects I did something like this:

```python
elastic_config = {"user": os.environ["ES_USER"],
                  "endpoint": os.environ["ES_ENDPOINT"],
                  ...
                  }
```

When I checked in the code I had been working on one of the reviewers commented
that this pattern was normal, but since we were using 3.7 let's use a
`dataclass`.

```python
import os
from dataclasses import dataclass

@dataclass
class ElasticConfiguration:
    user: str = os.environ["ES_USER"]
    endpoint: str = os.environ["ES_ENDPOINT"]
    ...
```

Makes sense, but what's the practical benefit? Before I wasn't defining a class
and carrying around the class model that I'm not really using.

1. Class attribute autocomplete. I can't tell you how many times I used to check
 if I had the right `_`, casing, abbreviation etc for the key I was calling. Now
 it's a class attribute, no more guessing.
2. Hook up `mypy` and find some interesting errors.
    1. Above you'll notice I used `os.environ[]`. A lot of people like to use an
     alternative `.get(<key>)` pattern with dictionaries. The problem is often
     times a default of None gets supplied and you're dealing with Optional[T],
     but still acting like it's str everywhere in your code.
3. `__post_init__`
    1. Dataclasses have an interesting method called
    [`__post_init__`](https://docs.python.org/3/library/dataclasses.html#post-init-processing)
    that gets called by `__init__`. On configuration objects this is a handy
    place to put any validation function/method calls you might build around
    attributes.
4. Subjectively `elastic.user` is faster to type, and more appealing to the eyes
 than `elastic["user"]`.

So the next time you find yourself passing around configuration information
remember dataclasses may be a useful and productive alternative to passing
around a dictionary.

### Additional Resources

Beyond the docs here are some links I found useful when learning about
Python dataclasses.

- [Real Python: Dataclasses](https://realpython.com/python-data-classes/)
- [Stack Overflow: What's the difference between a class and data class](https://stackoverflow.com/questions/47955263/what-are-data-classes-and-how-are-they-different-from-common-classes)
- [Hackernoon: Dataclasses tour](https://hackernoon.com/a-brief-tour-of-python-3-7-data-classes-22ee5e046517)

### Contact

If you want to reach out feel free to follow up via
[email](mailto:alexander@unexpextedeof.net) or on
[Sourcehut](https://git.sr.ht/%7En0mn0m/).
```

**Comment by n0mn0m** [comment]
As a side note this happens when I put the style in my `.mdlrc` too.

```
ignore_front_matter true
style "mdl.rb"
```

https://github.com/markdownlint/markdownlint/blob/master/docs/configuration.md

Without style set in `.mdlrc`


```bash
% mdl -c=.mdlrc input/posts/dataclasses-as-config-objects.d
input/posts/dataclasses-as-config-objects.md:57: MD029 Ordered list item prefix
input/posts/dataclasses-as-config-objects.md:62: MD029 Ordered list item prefix
input/posts/dataclasses-as-config-objects.md:68: MD029 Ordered list item prefix

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

With style set in my `.mdlrc`

```bash
% mdl -c=.mdlrc input/posts/dataclasses-as-config-objects.md
input/posts/dataclasses-as-config-objects.md:1: MD041 First line in file should be a top level header
input/posts/dataclasses-as-config-objects.md:29: MD046 Code block style
input/posts/dataclasses-as-config-objects.md:40: MD046 Code block style

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

**Comment by n0mn0m** [comment]
This turned out to be incorrectly formatted front_matter. Once I figured that out things started working as expected.
